cmd/agent/install_script.sh

```diff
@@ -235,8 +235,8 @@ fi
 restart_cmd="$sudo_cmd systemctl restart datadog-agent.service"
 stop_instructions="$sudo_cmd systemctl stop datadog-agent"
 start_instructions="$sudo_cmd systemctl start datadog-agent"
-# Upstart
-if command -v start >/dev/null 2>&1; then
+# Try to detect Upstart, this works most of the times but still a best effort
+if [[ `/sbin/init --version` =~ upstart ]]; then
```

Comment:
I'm not sure if the version is printed to `stdout` or `stderr`, but on `systemd` that'll print a nasty (albeit innocuous) message... Wondering if `/sbin/init --version 2>/dev/null` might work better (only if the output of `--version` goes to stdout)?

Reply:
Also `[[ ... ]]` is a bashism... It's not a huge deal as this is indeed a bash script, but we tried to keep away from them in the rest of the script I believe.

Reply:
even worse, it uses regexes, didn't think about this I might need to reconsider